File: app/src/main/java/com/ysdemo/mykotlin/ui/vlayout/SubAdapter.java
package com.ysdemo.mykotlin.ui.vlayout;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import com.alibaba.android.vlayout.DelegateAdapter;
import com.alibaba.android.vlayout.LayoutHelper;
import com.alibaba.android.vlayout.VirtualLayoutManager;
import com.ysdemo.mykotlin.R;

public class SubAdapter extends DelegateAdapter.Adapter<DemoViewHolder> {

        private Context mContext;

        private LayoutHelper mLayoutHelper;


        private VirtualLayoutManager.LayoutParams mLayoutParams;
        private int mCount = 0;


        public SubAdapter(Context context, LayoutHelper layoutHelper, int count) {
            this(context, layoutHelper, count, new VirtualLayoutManager.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 300));
        }

        public SubAdapter(Context context, LayoutHelper layoutHelper, int count, @NonNull VirtualLayoutManager.LayoutParams layoutParams) {
            this.mContext = context;
            this.mLayoutHelper = layoutHelper;
            this.mCount = count;
            this.mLayoutParams = layoutParams;
        }

        @Override
        public LayoutHelper onCreateLayoutHelper() {
            return mLayoutHelper;
        }

        @Override
        public DemoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            return new DemoViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item, parent, false));
        }

        @Override
        public void onBindViewHolder(DemoViewHolder holder, int position) {
            // only vertical
            holder.itemView.setLayoutParams(
                    new VirtualLayoutManager.LayoutParams(mLayoutParams));
        }


        @Override
        protected void onBindViewHolderWithOffset(DemoViewHolder holder, int position, int offsetTotal) {
            ((TextView) holder.itemView.findViewById(R.id.title)).setText(Integer.toString(offsetTotal));
        }

        @Override
        public int getItemCount() {
            return mCount;
        }
    }

File: app/src/main/java/com/ysdemo/mykotlin/ui/vlayout/SubAdapter.kt
package com.ysdemo.mykotlin.ui.vlayout

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView

import com.alibaba.android.vlayout.DelegateAdapter
import com.alibaba.android.vlayout.LayoutHelper
import com.alibaba.android.vlayout.VirtualLayoutManager
import com.ysdemo.mykotlin.R

open class SubAdapter @JvmOverloads constructor(private val mContext: Context, private val mLayoutHelper: LayoutHelper, count: Int, private val mLayoutParams: VirtualLayoutManager.LayoutParams = VirtualLayoutManager.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 300)) : DelegateAdapter.Adapter<DemoViewHolder>() {
    private var mCount = 0

    init {
        this.mCount = count
    }

    override fun onCreateLayoutHelper(): LayoutHelper {
        return mLayoutHelper
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DemoViewHolder {
        return DemoViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item, parent, false))
    }

    override fun onBindViewHolder(holder: DemoViewHolder, position: Int) {
        // only vertical
        holder.itemView.layoutParams = VirtualLayoutManager.LayoutParams(mLayoutParams)
    }


    override fun onBindViewHolderWithOffset(holder: DemoViewHolder?, position: Int, offsetTotal: Int) {
        (holder?.itemView?.findViewById(R.id.title) as? TextView)?.text = Integer.toString(offsetTotal)
    }

    override fun getItemCount(): Int {
        return mCount
    }
}
